x/collection/keeper/msg_server_test.go

```diff
@@ -1263,18 +1386,36 @@ func (s *KeeperTestSuite) TestMsgOperatorBurnFT() {
 			}
 
 			s.Require().NotNil(res)
-
-			if s.deterministic {
-				s.Require().Equal(tc.events, ctx.EventManager().Events())
-			}
+			s.Require().Equal(tc.events, ctx.EventManager().Events())
 		})
 	}
 }
 
 func (s *KeeperTestSuite) TestMsgBurnNFT() {
-	tokenIDs := []string{
-		collection.NewNFTID(s.nftClassID, s.numNFTs*2+1),
+	tokenID := collection.NewNFTID(s.nftClassID, s.numNFTs*2+1)
+	target := []string{tokenID}
+	tokenIDs := make([]string, 0)
+	tokenIDs = append(tokenIDs, tokenID)
+	cursor := tokenIDs[0]
+	for {
+		ctx, _ := s.ctx.CacheContext()
+		res, err := s.queryServer.Children(sdk.WrapSDKContext(ctx), &collection.QueryChildrenRequest{
+			ContractId: s.contractID,
+			TokenId:    cursor,
+			Pagination: &query.PageRequest{},
+		})
+		s.Require().NoError(err)
+		if res.Children == nil {
+			break
+		}
+		tokenIDs = append(tokenIDs, res.Children[0].TokenId)
+		cursor = tokenIDs[len(tokenIDs)-1]
 	}
+	coins := make([]collection.Coin, 0)
+	for _, id := range tokenIDs {
+		coins = append(coins, collection.NewCoin(id, sdk.NewInt(1)))
+	}
```

Comment:
Is this for making expectedTokenIDs and expectedAmount?
You may extract this block as a method with a descriptive name then it would be more readable.